### ISSUE TITLE
Cyrillic characters

### ISSUE THREAD

**Issue by wwwklopar** [TurtleP/LovePotion#57]
Hello. Looks like Lua can't support cyrillic characters from 1040 in unicode. Any way to fix this? I got too many troubles with this cause can't do translation for one of games on LovePotion without very many workarounds.
P.S. Thanks for your work!

**Comment by NotQuiteApex** [comment]
Lua supports any character formatting, ASCII or UTF-8, Lua doesn't mess with anything inside strings besides copying them into memory. Two things could be happening here: 1) the font in use does not have support for those characters or 2) the characters you're trying to use aren't at the correct index to match the character data in the string. I'm gonna guess it's the first thing, so try a font that supports those characters.

**Comment by wwwklopar** [comment]
Also i can attach fonts folder that i use. You can check them.
[Font](https://drive.google.com/file/d/11mLN3Zn0ApGoLARHbKUMbivoRLXQcpvx/view?usp=sharing)

**Comment by TurtleP** [comment]
There shouldn't be any issue on Switch. 3DS has some issue with loading json sometimes, but other than that I know I added unicode support to it.

**Comment by wwwklopar** [comment]
> There shouldn't be any issue on Switch. 3DS has some issue with loading json sometimes, but other than that I know I added unicode support to it.

On 3DS thing even better) If on Switch russian text transforms in something mess, then on 3DS i'm just have empty text field. [Photo](https://imgur.com/a/f8WsH42)

**Comment by wwwklopar** [comment]
If i for example I will change places russian and latin symbols then and will write text on translit then cyrillic symbols will work but then i must rewrite all text on translit and that's not good, because sometimes i can do errors with this work. I'm just don't know then whats the trouble.

**Comment by TurtleP** [comment]
If you’re using the latest release from this repository, you should be using a JSON file with png atlas for fonts. I thought I had it on the wiki, but anyways you need to use [this](http://www.angelcode.com/products/bmfont/) and then a program to export the bin file to JSON. I used [bmfont2json](https://www.npmjs.com/package/bmfont2json) for that.

**Comment by wwwklopar** [comment]
> If you’re using the latest release from this repository, you should be using a JSON file with png atlas for fonts. I thought I had it on the wiki, but anyways you need to use [this](http://www.angelcode.com/products/bmfont/) and then a program to export the bin file to JSON. I used [bmfont2json](https://www.npmjs.com/package/bmfont2json) for that.

Thats for 3DS?  If yes, than i already tried this with Hiero jar... , but result still the same. 
Russian text is not displayed in the game.
On switch game uses .ttf fonts. Will  .png and .json stuff work there as on 3DS?
[Fonts png] (https://imgur.com/a/OHevq31)

**Comment by TurtleP** [comment]
Yes on 3ds it’s png and JSON. They should work, although the JSON library sometimes has issues and I don’t know why.

**Comment by wwwklopar** [comment]
> Yes on 3ds it’s png and JSON. They should work, although the JSON library sometimes has issues and I don’t know why.

Well then i don't know what's the trouble cause i already tried to change .png and .json
Can you tell what settings are you using? I will try your programs for test

**Comment by TurtleP** [comment]
@wwwklopar have you managed to fix the issue?

**Comment by wwwklopar** [comment]
> @wwwklopar have you managed to fix the issue?

Well) Cyrillic characters are still not works, but i rework fonts with 
replacing latin symbols to cyrillic and wrote special text so now translation almost ready.
On 3DS same way. Maybe i'm doing something wrong but i managed to translate only that way.

**Comment by TurtleP** [comment]
That's really odd, honestly. Switch shouldn't have any issues loading a Cyrillic font file (as long as it's a ttf, not sure of other formats). However, 3DS I understand that you had to do something different, good to know you found a solution :)

**Comment by wwwklopar** [comment]
On Switch i have problems with Cyrillic even in html... Like when you create custom nsp video player.
Thanks for your work! Happy New Year!
